packages/devextreme-schematics/src/add-layout/index.ts

```diff
@@ -250,13 +273,25 @@ function hasRouting(host: Tree, sourcePath: string) {
 
 function addPackagesToDependency(globalNgCliVersion: string) {
   const version = new SemVer(globalNgCliVersion.replace(/\^|\~/g, ''));
+
   return (host: Tree) => {
+    const zonejs = getPackageJsonDependency(host, 'zone.js');
+
     addPackageJsonDependency(host, {
       type: NodeDependencyType.Default,
       name: '@angular/cdk',
-      version: `~${version.major}.${version.minor}.0`
+      version: version.raw.includes('next') || version.raw.includes('-rc')
+        ? 'next' : `^${version.major}.${version.minor > 1 ? version.minor - 1 : 0}.0`
```

Comment:
So, `@angular/cli` and `@angular/cdk` have different minor versioning, and we have here `version.minor - 1` because usually `@angular/cdk` has at max one minor release less...

And if, some times, Angular releases `22.4.1` CLI and only `22.2.1` CDK this won't work again?

An easy way would be to just zero the minor for CDK `^major.0.0` to let it install the latest minor, but there will be issues if the local CLI is not latest, right?

Although, who knows if they are not in sync...

Reply:
yes. There is no 100% guarantee in this case. This is just a more general solution than before, and it also handles the case I caught